internal/service/userService.go

```diff
@@ -0,0 +1,69 @@
+package service
+
+import (
+	"context"
+	"errors"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/internal/repository"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/internal/service/models"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/internal/service/models/exceptions"
+	logger "github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/topology"
+)
+
+type UserService interface {
+	SaveUser(ctx context.Context, user models.User) (*models.User, error)
+	GetUserById(ctx context.Context, id string) (*models.User, error)
+}
+
+type service struct {
+	userRepository repository.UserRepository
+}
+
+func NewUserService(repository repository.UserRepository) UserService {
+	return &service{
+		userRepository: repository,
+	}
+}
+
+const errorMessage = "something went wrong"
+
+func (srv *service) SaveUser(ctx context.Context, user models.User) (*models.User, error) {
+	logger.Infof("Saving user %s", user.GetFullName())
+	userByFullName, err := srv.userRepository.GetUserByFullName(ctx, user.GetFirstName(), user.GetLastName())
+
+	if userByFullName != nil {
+		if err == nil {
+			logger.Errorf("User already exists: %v", user.GetFullName())
+			return nil, nil
+		}
+		logger.Errorf("Something went wrong: %v", err)
```

Comment:
You could use fmt.Errorf or errors.Join to merge errors together, this allows you to log all errors in a cascade, indicating what service error happened and that the root cause was, in this case that you user already exists. And you could also compare the error response on a test or validation using errors.Is